[tjch-o] iP

tjch-o opened the pull request:
# Miles
> "You were right, Gwen. You should have never come to see me." - [Miles Morales](https://intothespiderverse.fandom.com/wiki/Miles_Morales), 2023

- text-based 
- easy-to-learn
- ~FAST~ LIGHT SPEED to use

All you need to do is:
1. download it from [here](https://intothespiderverse.fandom.com/wiki/Gwen_Stacy_(Earth-65))
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Managing reminders (coming soon)

If you Java programmer,  you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tjch-o committed:
change chatbot name and add rename, greet, exit feature

tjch-o committed:
add echo feature

tjch-o committed:
implement add, list features

tjch-o committed:
add mark as done and change status back to undone feature

tjch-o committed:
add ToDos, Events, Deadlines features

tjch-o committed:
add automated ui testing

tjch-o committed:
add chatbot error handling ability for invalid user input

tjch-o committed:
add delete feature and improve on automated ui testing

tjch-o committed:
added more comprehensive testing

tjch-o committed:
Implement save feature

tjch-o committed:
Add minor modifications to save feature

tjch-o committed:
Merge branch 'branch-Level-7'

tjch-o committed:
Implement DateTime for deadlines and events

tjch-o committed:
Merge branch 'branch-Level-8'

tjch-o committed:
Refactor code to add more OOP

tjch-o committed:
Merge branch 'A-MoreOOP'

tjch-o committed:
Add packages and reorganise files into correct folders

tjch-o committed:
Merge branch 'branch-A-Packages'

tjch-o committed:
Merge branch 'add-gradle-support' of https://github.com/tjch-o/ip into add-gradle-support

tjch-o committed:
Merge branch 'add-gradle-support'

tjch-o committed:
Add first method JUnit test for Event class

tjch-o committed:
Add JUnit tests for Storage class

tjch-o committed:
Add more JUnit tests for Event class

tjch-o committed:
Add JUnit tests for Parser class

tjch-o committed:
Merge branch 'branch-A-JUnit'

tjch-o committed:
Update JUnit test cases

tjch-o committed:
Fix directory cannot be found bug for the Storage JUnit test

tjch-o committed:
Add JavaDocs to most of the classes

tjch-o committed:
Improve code to follow coding standard

tjch-o committed:
Implement find feature

tjch-o committed:
Merge branch 'branch-A-JavaDoc'

tjch-o committed:
Merge branch 'branch-A-CodingStandard'

tjch-o committed:
Merge branch 'branch-Level-9'

tjch-o committed:
Rename folders to reflect correct name of chatbot

tjch-o committed:
Refactor code to abstract out Command

chonguschonguschongus commented on src/main/java/miles/command/Command.java:
```diff
@@ -0,0 +1,39 @@
+package miles.command;
```
Good job extracting out Command 👍

chonguschonguschongus commented on src/main/java/miles/Ui.java:
```diff
@@ -0,0 +1,98 @@
+     * Greets the user.
+     */
+    public void greet() {
+        String chatbotName = "Miles";
```
What's the point of having an extra variable here? Can't the method just print Hi I am Miles? 😢

chonguschonguschongus commented on src/main/java/miles/task/Task.java:
```diff
@@ -0,0 +1,104 @@
+package miles.task;
```
Nice job extracting Task class 👍

chonguschonguschongus reviewed (commented):
LGTM. Everything looks great in terms of coding standard and general OOP

tjch-o committed:
Incorporate checkstyle

tjch-o committed:
Fix minor checkstyle errors

tjch-o committed:
Merge branch 'branch-A-CheckStyle'

tjch-o committed:
Improve javadocs

tjch-o committed:
Merge branch 'branch-A-CheckStyle' into branch-Level-10

tjch-o committed:
Implement GUI for chatbot

tjch-o committed:
Correct controller in MainWindow.fxml

tjch-o committed:
Minor adjustments to GUI

tjch-o committed:
Fix bug where GUI cannot find task list and improve on GUI

tjch-o committed:
Merge branch 'branch-Level-10'

tjch-o committed:
Refactor code

tjch-o committed:
Fix list being truncated issue

alientian commented on src/main/java/miles/task/Task.java:
```diff
@@ -0,0 +1,104 @@
+     * 
+     * @param taskType  the type of task (e.g. "deadline")
+     * @param task      the command inputted by the user
+     * @return          a boolean that indicates whether the task has no description
```
Its a little confusing what the boolean value means 
```suggestion
     * @return          true if the task has no description
```

alientian commented on src/main/java/miles/task/Event.java:
Very comprehensive documentation! 👍

alientian commented on src/main/java/miles/command/MarkCommand.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    /**
+     * Return task number from user input.
```
Just a minor problem, JavaDoc comments should start with Returns ... 
and include punctuation behind each parameter description.
Mistake is repeated in other files too, remember to keep a look out!

alientian reviewed (commented):
Looks good! Naming of variables and methods are really good!

tjch-o committed:
Fix unnecessary spaces bug for user dialog and remove unnecessary indentations due to changing code requirements

tjch-o committed:
Add assertions to the codebase

This is needed because we want to avoid regressions to the software as
more features are being added.

Add assertions to the tasklist to check that the index is valid before
we mark, unmark or delete a task and also to the todo, deadline and
event tasks to check if there are the right number of elements in the
arrays after splitting before further formatting can be done.

This ensures more robustness in the codebase.

tjch-o committed:
Improve quality of codebase by following coding standards

This ensures that code is more readable and maintainable.

Change codebase to remove unnecessary comments and also refactor code to
reduce level of nesting. Some whitespaces are also added to provide a
logical structure to the code.

tjch-o committed:
Merge pull request #1 from tjch-o/branch-A-Assertions

Add assertions to the codebase

tjch-o committed:
Merge branch 'master' into branch-A-CodeQuality

tjch-o committed:
Merge pull request #2 from tjch-o/branch-A-CodeQuality

Improve quality of codebase

tjch-o committed:
Remove legacy code and fix exception handling in the GUI

This ensures that users will be able to troubleshoot when using the chat
bot.

Changed all the IllegalArgumentException to MilesException and also
change the method name from `formatString` to `display` to improve code
readability.

tjch-o committed:
Implement detect duplicates feature

tjch-o committed:
Merge branch 'branch-C-DetectDuplicates'

tjch-o committed:
Add continuous integration

tjch-o committed:
Implement handling of invalid datetime formats

tjch-o committed:
Merge branch 'branch-A-CI' of https://github.com/tjch-o/ip into branch-A-CI

tjch-o committed:
Fix checkstyle errors

tjch-o committed:
Improve checkstyle on Junit tests

tjch-o committed:
Fix checkstyle error

tjch-o committed:
Fix checkstyle errors

tjch-o committed:
Merge branch 'branch-A-CI'

tjch-o committed:
Fix checkstyle errors

tjch-o committed:
Fix checkstyle errors detected during building with gradle

tjch-o committed:
Implement more thorough error handling

tjch-o committed:
Remove legacy code

tjch-o committed:
Update test cases affected by changes to codebase

tjch-o committed:
Fix minor checkstyle error

tjch-o committed:
Merge branch 'branch-A-MoreErrorHandling'

tjch-o committed:
Update user guide

tjch-o committed:
Perform minor adjustments to code

tjch-o committed:
Fix program not exiting error

tjch-o committed:
Fix bug when an invalid input is given to the mark or unmark command
